src/site/content/en/blog/promises/index.md

```diff
@@ -0,0 +1,1210 @@
+settle:
+
+```js
+function spawn(generatorFunc) {
+  function continuer(verb, arg) {
+    var result;
+    try {
+      result = generator[verb](arg);
+    } catch (err) {
+      return Promise.reject(err);
+    }
+    if (result.done) {
+      return result.value;
+    } else {
+      return Promise.resolve(result.value).then(onFulfilled, onRejected);
+    }
+  }
+  var generator = generatorFunc();
+  var onFulfilled = continuer.bind(continuer, "next");
+  var onRejected = continuer.bind(continuer, "throw");
+  return onFulfilled();
+}
+```
+
+… which I pretty much
+[lifted verbatim from Q](https://github.com/kriskowal/q/blob/db9220d714b16b96a05e9a037fa44ce581715e41/q.js#L500),
+but adapted for JavaScript promises. With this, we can take our final
+best-case chapter example, mix it with a load of new ES6 goodness, and turn
+it into:
+
+```js
+spawn(function *() {
+  try {
+    // 'yield' effectively does an async wait,
+    // returning the result of the promise
+    let story = yield getJSON('story.json');
+    addHtmlToPage(story.heading);
+
+    // Map our array of chapter urls to
+    // an array of chapter json promises.
+    // This makes sure they all download in parallel.
+    let chapterPromises = story.chapterUrls.map(getJSON);
+
+    for (let chapterPromise of chapterPromises) {
+      // Wait for each chapter to be ready, then add it to the page
+      let chapter = yield chapterPromise;
+      addHtmlToPage(chapter.html);
+    }
+
+    addTextToPage("All done");
+  }
+  catch (err) {
+    // try/catch just works, rejected promises are thrown here
+    addTextToPage("Argh, broken: " + err.message);
+  }
+  document.querySelector('.spinner').style.display = 'none';
+})
+```
+
+<div class="glitch-embed-wrap" style="height: 480px; width: 100%;">
+  <iframe src="https://glitch.com/embed/#!/embed/promises-async-generator-example?previewSize=100"
+          alt="An example of fetching JSON asynchronously with a generator."
+          style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+This works exactly as before but is so much easier to read. This works in Chrome
+and Opera today, and works in Microsoft Edge by going to `about:flags` and
+turning on the **Enable experimental JavaScript features** setting. This will be
+enabled by default in an upcoming version.
+
+This throws together a lot of new ES6 stuff: promises, generators, let, for-of.
+When we yield a promise, the spawn helper waits for the promise to resolve and
+returns the final value. If the promise rejects, spawn causes our yield
+statement to throw an exception, which we can catch with normal JavaScript
+try/catch. Amazingly simple async coding!
+
+This pattern is so useful, it's coming to ES7 in the form of
+[async functions](https://jakearchibald.com/2014/es7-async-functions/). It's
+pretty much the same as above, but no need for a `spawn` method.
+
+## Promise API reference {: #promise-api-reference }
```

Comment:
I'd remove this section and just link to MDN.

Reply:
Agreed but for now I'm just doing a minimal migration of the content.